### PR TITLE
ddl: add more tests for expression index

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

add more tests for expression index

### What is changed and how it works?

add more tests for expression index

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
 'None'
```

### CODE REVIEW

File: ddl/db_change_test.go [hunk 1]
```diff
@@ -1884,10 +1920,259 @@ func (s *serialTestStateChangeSuite) TestCreateExpressionIndex(c *C) {
...
+			}
+			_, checkErr = tk1.Exec("begin;")
+			if checkErr != nil {
+				return
+			}
+			_, tmpErr := tk1.Exec("insert into t select * from t")
+			if tmpErr == nil {
+				checkErr = errors.New("should not be nil")
+			}
+			_, checkErr = tk1.Exec("rollback")
```
Comment (zimulala) [inline ddl/db_change_test.go:1995]: This line value of `checkErr` maybe overwrite the value in line1993.

File: ddl/db_change_test.go [hunk 2]
```diff
@@ -1884,10 +1920,259 @@ func (s *serialTestStateChangeSuite) TestCreateExpressionIndex(c *C) {
...
+			}
+			_, checkErr = tk1.Exec("update t set b = 7 where a = 1")
+			if checkErr != nil {
+				return
+			}
+			_, checkErr = tk1.Exec("delete from t where b = 4")
+			if checkErr != nil {
+				return
+			}
+			// (1, 7), (2, 7), (5, 5), (8, 8), (0, 9), (0, 6)
```
Comment (zimulala) [inline ddl/db_change_test.go:2105]: Feeling a problem with this result? Where the `(3, 3)`?

File: ddl/db_change_test.go [hunk 3]
```diff
@@ -1884,10 +1920,259 @@ func (s *serialTestStateChangeSuite) TestCreateExpressionIndex(c *C) {
...
+			}
+			_, checkErr = tk1.Exec("update t set b = 7 where a = 2")
+			if checkErr != nil {
+				return
+			}
+			_, checkErr = tk1.Exec("delete from t where b = 3")
+			if checkErr != nil {
+				return
+			}
+			// (1, 1), (2, 7), (4, 4), (8, 8), (0, 9)
```
Comment (zimulala) [inline ddl/db_change_test.go:2135]: Ditto.

File: ddl/db_change_test.go [hunk 4]
```diff
@@ -1826,25 +1830,88 @@ func (s *serialTestStateChangeSuite) TestCreateExpressionIndex(c *C) {
...
+				}
+			}
+			// (1, 7), (2, 7), (5, 7), (8, 8), (0, 9), (10, 10), (10, 10), (0, 11), (0, 11)
+		}
+	}
+
+	d.(ddl.DDLForTest).SetHook(callback)
+	tk.MustExec("alter table t add index idx((b+1))")
+	c.Assert(checkErr, IsNil)
+	tk.MustExec("admin check table t")
```
Comment (AilinKid) [inline ddl/db_change_test.go:1862]: how about adding `admin check table t` at the end of each state.

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

File: ddl/db_change_test.go (hunk 4 above)
Comment (wjhuang2016) [reply ddl/db_change_test.go:1862]: It's useless because the index is not public yet.

**Review by zimulala — APPROVED** [review]
LGTM

**Review by AilinKid — APPROVED** [review]

**Pull request merged**